### PR TITLE
Support for configurable logger

### LINKED ISSUE #365: Support user-defined logger instance
First off, thank you for creating the `go-ldap` library and sharing it with the community. 

Using the optional debug logger was helpful while integrating the library. When transitioning from dev/test to a prod version of an application, the debug logger can be disabled, which is super convenient. If a project that imports the library uses a custom logger instance, the library has a fallback to the default `log.Logger` instance in certain scenarios, which log the messages to stderr and may not be ideal. It would be helpful to inject a `log.Logger` instance into the library to handle these cases in a way that is consistent with the rest of the application.

If this is functionality that you think that others in the community would benefit from, I can submit a PR.

### PR DESCRIPTION
**Pull request by n3integration**
Fixes #365 

Initial pass at a pull request to introduce an `ldap.logger` variable that defaults to the `log.Default()` logger (to preserve backwards compatibility), but can be overridden at runtime through the `ldap.Logger(l *log.Logger)` function.

### CODE REVIEW

File: v3/ldap.go [hunk 1]
```diff
@@ -82,6 +83,13 @@ var BeheraPasswordPolicyErrorMap = map[int8]string{
 	BeheraPasswordInHistory:           "New password is in list of old passwords",
 }
 
+var logger = log.Default()
+
+// Logger allows clients to override the default logger
+func Logger(l *log.Logger) {
+	logger = l
+}
```
Comment (johnweldon) [inline v3/ldap.go:91]: Out of curiosity; why a private variable and helper function vs a public variable like the stdlib does?

**Comment by johnweldon** [comment]
We need to fix the validators too

File: v3/ldap.go (hunk 1 above)
Comment (johnweldon) [reply v3/ldap.go:91]: I can't find the example in the stdlib I was thinking of... ignore this :)

File: v3/ldap.go [hunk 2]
```diff
@@ -82,6 +83,13 @@ var BeheraPasswordPolicyErrorMap = map[int8]string{
 	BeheraPasswordInHistory:           "New password is in list of old passwords",
 }
 
+var logger = log.Default()
```
Comment (johnweldon) [inline v3/ldap.go:86]: so we can support Go v1.14 and v1.15, what do you think about this:
```suggestion
var logger = log.New(os.Stderr, "", log.LstdFlags)
```

File: v3/ldap.go (hunk 1 above)
Comment (n3integration) [reply v3/ldap.go:91]: Great question. Encapsulation. Making the logger private to the package, it cannot be directly used (or abused) outside of the library.

File: v3/ldap.go (hunk 2 above)
Comment (n3integration) [reply v3/ldap.go:86]: no worries; that works 👍

**Comment by johnweldon** [comment]
Sorry, should have specified that we should do it in the other file too - we're still trying to keep them in sync

File: ldap.go [hunk 3]
```diff
@@ -82,6 +83,13 @@ var BeheraPasswordPolicyErrorMap = map[int8]string{
 	BeheraPasswordInHistory:           "New password is in list of old passwords",
 }
 
+var logger = log.Default()
```
Comment (n3integration) [inline ldap.go:86]:
```suggestion
var logger = log.New(os.Stderr, "", log.LstdFlags)
```

**Pull request merged**